pkgs/development/python-modules/monty/default.nix

```diff
@@ -2,81 +2,117 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
-  msgpack,
-  numpy,
-  pandas,
-  pydantic,
-  pymongo,
-  pytestCheckHook,
   pythonOlder,
-  ruamel-yaml,
+
+  # build-system
   setuptools,
   setuptools-scm,
+
+  # dependencies
+  msgpack,
+  ruamel-yaml,
+
+  # optional-dependencies
+  coverage,
+  pymongo,
+  pytest,
+  pytest-cov,
+  types-requests,
+  sphinx,
+  sphinx-rtd-theme,
+  orjson,
+  pandas,
+  pydantic,
+  pint,
   torch,
   tqdm,
+  invoke,
+  requests,
+
+  # tests
+  ipython,
+  numpy,
+  pytestCheckHook,
 }:
 
 buildPythonPackage rec {
   pname = "monty";
   version = "2025.1.9";
   pyproject = true;
 
-  disabled = pythonOlder "3.9";
-
   src = fetchFromGitHub {
     owner = "materialsvirtuallab";
     repo = "monty";
     tag = "v${version}";
     hash = "sha256-+9nxuvrtl04Fb+GQJUn+fxg4sq1pXK59UkUTkKH16YQ=";
   };
 
-  postPatch = ''
-    substituteInPlace tests/test_os.py \
-      --replace 'self.assertEqual("/usr/bin/find", which("/usr/bin/find"))' '#'
-  '';
-
-  nativeBuildInputs = [
+  build-system = [
     setuptools
     setuptools-scm
   ];
 
-  propagatedBuildInputs = [
+  dependencies = [
     msgpack
     ruamel-yaml
-    tqdm
   ];
 
+  optional-dependencies = rec {
+    ci = [
+      coverage
+      pymongo
+      pytest
+      pytest-cov
+      types-requests
+    ] ++ optional;
+    dev = [ ipython ];
+    docs = [
+      sphinx
+      sphinx-rtd-theme
+    ];
+    json =
+      [
+        orjson
+        pandas
+        pydantic
+        pymongo
+      ]
+      ++ lib.optionals (pythonOlder "3.13") [
+        pint
+        torch
+      ];
+    multiprocessing = [ tqdm ];
+    optional = dev ++ json ++ multiprocessing ++ serialization;
+    serialization = [ msgpack ];
+    task = [
+      invoke
+      requests
+    ];
+  };
+
   nativeCheckInputs = [
+    ipython
     numpy
     pandas
     pydantic
     pymongo
     pytestCheckHook
     torch
+    tqdm
   ];
 
   pythonImportsCheck = [ "monty" ];
 
-  disabledTests = [
-    # Test file was removed and re-added after 2022.9.9
-    "test_reverse_readfile_gz"
-    "test_Path_objects"
-    "test_zopen"
-    "test_zpath"
-    # flaky, precision/rounding error
-    "TestJson.test_datetime"
-  ];
-
-  meta = with lib; {
+  meta = {
     description = "Serves as a complement to the Python standard library by providing a suite of tools to solve many common problems";
     longDescription = "
       Monty implements supplementary useful functions for Python that are not part of the
       standard library. Examples include useful utilities like transparent support for zipped files, useful design
       patterns such as singleton and cached_class, and many more.
     ";
     homepage = "https://github.com/materialsvirtuallab/monty";
-    changelog = "https://github.com/materialsvirtuallab/monty/releases/tag/${src.tag}";
```

Comment:
I'd prefer to keep this as is until there's consensus but if you hate referencing `src` it's fine with me.

Reply:
The thing is that `meta` is irrelevant when overriding. Anyway, this is not the place to discuss this.